xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -1737,27 +1723,40 @@ void start() {
       StreamObserver<io.envoyproxy.envoy.api.v2.DiscoveryResponse> responseReaderV2 =
           new StreamObserver<io.envoyproxy.envoy.api.v2.DiscoveryResponse>() {
             @Override
-            public void onNext(io.envoyproxy.envoy.api.v2.DiscoveryResponse response) {
-              DiscoveryResponseData responseData =
-                  DiscoveryResponseData.fromEnvoyProtoV2(response);
-              if (logger.isLoggable(XdsLogLevel.DEBUG)) {
-                logger.log(
-                    XdsLogLevel.DEBUG,
-                    "Received {0} response:\n{1}",
-                    responseData.getResourceType(),
-                    respPrinter.print(response));
-              }
-              onDiscoveryResponse(responseData);
+            public void onNext(final io.envoyproxy.envoy.api.v2.DiscoveryResponse response) {
+              syncContext.execute(new Runnable() {
+                @Override
+                public void run() {
+                  if (logger.isLoggable(XdsLogLevel.DEBUG)) {
+                    logger.log(
+                        XdsLogLevel.DEBUG,
+                        "Received DiscoveryResponse:\n{0}", respPrinter.print(response));
```

Comment:
Why change the logging message?

Reply:
Well, it's a small cosmetic change. Two reasons:
- Make this logging the first operation to be run in the handler, without needing information from parsed data. This seems a bit more elegant?
- Just log out the implementation types for DEBUG logs, indicating the content being logged is the actual code.

Reply:
You can still add resource type information, which is more user-friendly 
```
ResourceType.fromTypeUrl(response.getTypeUrl())
```

Reply:
That's a long URL, not good. Also, the raw response already contains everything, so not bother do anything excessive.

Reply:
>That's a long URL

It's converted to type enum.

>the raw response already contains everything,
That's long URL, it's hard to eyeball the type quickly.

Reply:
> It's converted to type enum.

Oh you mean converted parsed data. But as I said, it's better to just log everything raw.

> That's long URL, it's hard to eyeball the type quickly.

Actually it works pretty good from my experience, as people usually care more about `Listener`/`RouteConfiguration`/`Cluster`/`ClusterLoadAssignment` rather than xDS. IMO, it's better not to add anything excessive to raw data logs.

Reply:
Also, those are DEBUG level logs, they are intended to be read carefully when enabled for debugging, instead of quick eyeballing for what happens.

Reply:
> Oh you mean converted parsed data. But as I said, it's better to just log everything raw.

Not the parsed data, `ResourceType.fromTypeUrl(response.getTypeUrl())` is the raw data.

Reply:
Why do you want to decorate the raw data? Keep the plain data is concise.